src/extensions/default/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/PackageUpdaterStep.cs

```diff
@@ -150,6 +155,44 @@ void AddSubsteps<T>(IEnumerable<Operation<T>> items, string messageFormat, Func<
                 return new UpgradeStepInitializeResult(UpgradeStepStatus.Incomplete, $"{_analyzer.Name} has identified some recommended changes", Risk: _risk);
             }
 
+            // For server-only projects, do not need to add System.ServiceModel packages.
+            private static IEnumerable<Operation<NuGetReference>> UpdatePackageAddition(IDependencyCollection<NuGetReference> packages, IProject project)
+            {
+                var files = project.FindFiles(".cs", ProjectItemType.Compile);
+                var containsService = false;
+                if (files.Any())
```

Comment:
You can remove this `.Any()` check; the `foreach` loop won't execute if the collection is empty so this is basically making it do the work twice.